saveJsonFile.py

```diff
@@ -0,0 +1,131 @@
+
+try:
+    # For Python 3.0 and later
+    from urllib.error import HTTPError
+    from urllib.parse import quote
+    from urllib.parse import urlencode
+except ImportError:
+    # Fall back to Python 2's urllib2 and urllib
+    from urllib2 import HTTPError
+    from urllib import quote
+    from urllib import urlencode
+
+API_KEY= 'V4mhEirjIuAFM2xPFw9Zx3h-OccaRn079SLXXLPAtjk4yLLUB1v-hKRXLmqfPClCY0uQhIJSl6qhCKpAE7j9LVc14F15ufG4KmMMq-Xo0YidXVQqqiSHmXmSgwhQWnYx'
+
+# API constants, you shouldn't have to change these.
+API_HOST = 'https://api.yelp.com'
+SEARCH_PATH = '/v3/businesses/search'
+BUSINESS_PATH = '/v3/businesses/'  # Business ID will come after slash.
+
+# Defaults for our simple example.
+DEFAULT_TERM = 'cafe'
+DEFAULT_LOCATION = 'San Francisco, CA'
+SEARCH_LIMIT = 3
+
+def request(host, path, api_key, url_params=None):
+    """Given your API_KEY, send a GET request to the API.
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        API_KEY (str): Your API Key.
+        url_params (dict): An optional set of query parameters in the request.
+    Returns:
+        dict: The JSON response from the request.
+    Raises:
+        HTTPError: An error occurs from the HTTP request.
+    """
+    url_params = url_params or {}
+    url = '{0}{1}'.format(host, quote(path.encode('utf8')))
+    headers = {
+        'Authorization': 'Bearer %s' % api_key,
+    }
+
+    response = requests.request('GET', url, headers=headers, params=url_params)
+
+    return response.json()
+
+
+def search(api_key, term, location):
+    """Query the Search API by a search term and location.
+    Args:
+        term (str): The search term passed to the API.
+        location (str): The search location passed to the API.
+    Returns:
+        dict: The JSON response from the request.
+    """
+
+    url_params = {
+        'term': term.replace(' ', '+'),
+        'location': location.replace(' ', '+'),
+        'limit': SEARCH_LIMIT
+    }
+    return request(API_HOST, SEARCH_PATH, api_key, url_params=url_params)
+
+
+def get_business(api_key, business_id):
+    """Query the Business API by a business ID.
+    Args:
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+
+def query_api(term, location):
+    """Queries the API by the input values from the user.
+    Args:
+        term (str): The search term to query.
+        location (str): The location of the business to query.
+    """
+    response = search(API_KEY, term, location)
+
+    businesses = response.get('businesses')
+
+    if not businesses:
+        return
+
+    businessList = []
+    for idx in businesses:
+        response = get_business(API_KEY, idx['id'])
+        businessList.append(response)
+
+    with open("result.json", "w") as f:
```

Comment:
this method should only do api lookup, and return the results. Move this code block into main().

Reply:
Oh! It's good! Thank you!